eclipse-collections/src/main/java/org/eclipse/collections/impl/map/sorted/immutable/ImmutableTreeMap.java

```diff
@@ -351,12 +355,58 @@ public boolean containsAll(Collection<?> collection)
             return Iterate.allSatisfyWith(collection, Predicates2.in(), this);
         }
 
+        @Override
+        public void each(Procedure<? super K> procedure)
+        {
+            ArrayIterate.forEach(ImmutableTreeMap.this.keys, procedure);
+        }
+
         @Override
         public int size()
         {
             return ImmutableTreeMap.this.keys.length;
         }
 
+        @Override
+        public int indexOf(Object object)
+        {
+            return 0;
+        }
+
+        @Override
+        public K getFirst()
+        {
+            if (this.size() > 0)
+            {
+                return (K) this.toArray()[0];
+            }
+            return null;
+        }
+
+        @Override
+        public K getLast()
+        {
+            if (this.size() > 0)
+            {
+                return (K) this.toArray()[0];
```

Comment:
Use `IterableIterate.getLast(this)`.

Reply:
@donraab getLast will iterate over the whole collection, is that ok?